File: Python/longest-palindrome.py
class Solution:
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        d = dict()
        res = 0

        for char in s:
            if char in d:
                if d[char] == 1:
                    d[char] = 0
                    res += 2
                else:
                    d[char] = d[char] + 1
            else:
                d[char] = 1

        for value in d.values():
            if value > 0:
                res += 1
                break
        
        return res

File: Swift/longest-palindrome.swift
class Solution {
    func longestPalindrome(_ s: String) -> Int {
        var dict = [Character: Int]()
        var res = 0

        for char in s {
            if let count = dict[char] {
                if count == 1 {
                    dict[char] = 0
                    res += 2
                } else {
                    dict[char] = count + 1
                }
            } else {
                dict[char] = 1
            }
        }

        if dict.contains(where: {$0.value > 0 }) {
            res += 1
        }

        return res
    }
}
